### PR TITLE
Remove assert in sse_round_fn and handle case where src2 is in memory

### PR DESCRIPTION
**Pull request by sydhds**
In Singlepass SSE4.2 backend and for sse_round_fn function, some special cases were not handled:
* src1 != src2
* src2 is in memory

This should fix those cases.

This is a follow up of ticket https://github.com/wasmerio/wasmer/issues/3461

### CODE REVIEW

File: lib/compiler-singlepass/src/emitter_x64.rs [hunk 1]
```diff
@@ -909,11 +909,12 @@ macro_rules! sse_round_fn {
         |emitter: &mut AssemblerX64, precision: Precision, src1: XMM, src2: XMMOrMemory, dst: XMM| {
             match src2 {
                 XMMOrMemory::XMM(x) => {
-                    assert_eq!(src1, x);
-                    move_src_to_dst(emitter, precision, src1, dst);
+                    move_src_to_dst(emitter, precision, x, dst);
```
Comment (ptitSeb) [inline lib/compiler-singlepass/src/emitter_x64.rs:912]: No like this.
You need to create an `if src1 == x` with that new code, and have the old code in the `else` branch.

**Review by ptitSeb — CHANGES_REQUESTED** [review]
Please create a `if` /  `else` branch

**Review by ptitSeb — CHANGES_REQUESTED** [review]
So, I reviewed the ROUNDSS/VROUNDSS opcode (for example, it's one of the opcode use here): https://www.felixcloutier.com/x86/roundss
And it's not what I imagined.
So the code should be:
```rust
if x != dst {
 move_src_to_dst(emitter, precision, src1, dst);
}
dynasm!(emitter ; $ins Rx((x as u8)), Rx((dst as u8)), $mode)
```
Basicaly a bigger change, but that version should be exact. It's not important for now, as ROUND functions are basically used as 2 operators opcode, never as 3-way in Singlepass.

**Review by ptitSeb — APPROVED** [review]

**Review by syrusakbary — APPROVED** [review]
It should already be approved. But approving again to see if that trigger tests

**Comment by syrusakbary** [comment]
Merged manually, as I had no idea why tests are not being executed

**Comment by theduke** [comment]
@syrusakbary Github Actions have been problematic the last two days, also on other repos.
Timeouts, network errors, jobs just sitting idle forever, ...

**Pull request merged**